modules/telemetry/telemetry/tests/log_tests.cpp

```diff
@@ -18,11 +22,50 @@
 // NOLINTNEXTLINE(google-build-using-namespace)
 using namespace ::testing;
 
-namespace hephtelemetry::tests {
+namespace heph::telemetry::tests {
 
 namespace ht = heph::telemetry;
 
-TEST(log, LogEntry) {
+/// @brief Custom sink that will write the log to a string.
+class MockLogSink final : public ht::ILogSink {
+  using Line = size_t;
+
+public:
+  void send(const ht::LogEntry& s) override {
+    logs_.emplace(s.location.line(), fmt::format("{}", s));
+  }
+
+  [[nodiscard]] auto getLog(std::source_location loc) -> const std::string& {
+    /*
+    std::unique_lock<std::mutex> lock(mtx);
+    std::cout << "waiting for " << loc.line() << "\n";
+    cv.wait(lock, [&]() { return logs_.find(loc.line()) != logs_.cend(); });
+*/
+    using namespace std::chrono_literals;
+    while (logs_.find(loc.line()) == logs_.cend()) {
+      std::this_thread::sleep_for(10ms);
+    }
+    return logs_.at(loc.line());
+  }
+
+private:
+  std::map<Line, std::string> logs_;
```

Comment:
You can use a blocking queue. From the test it looks like you call `log` and then `getLog` so this should work. It will block until you have a log message available.
```suggestion
  void send(const ht::LogEntry& s) override {
    logs_.emplaceBack(fmt::format("{}", s));
  }

  [[nodiscard]] auto getLog() -> std::string {
      return logs_.waitAndPop().value();
  }

private:
    containers::BlockingQueue<std::string> logs_;
```

Reply:
aweseome!